src/Sarif.FunctionalTests/SarifConverterTests.cs

```diff
@@ -77,6 +78,32 @@ public void PREfastConverter_EndToEnd()
             BatchRunConverter(ToolFormat.PREfast);
         }
 
+        [Fact]
+        public void ValidateDeserializationAndPrettifiedSarifStringComparisionLogic()
```

Comment:
I appreciate your thinking about this, but I'm uncomfortable with this. In general, it's not necessary to test your test code (because where would you stop? would you test the code that tests your test code as well?) And in this particular case, you have to go to a lot of trouble -- four new test files and two new tests.

I would remove these tests and the new test files, and then I'd just do a quick hand verification of your logic as follows:

1. Run `BuildAndTest.cmd` -- this ensures that all the tests you expect to pass, do continue to pass.
2. Modify one of the functional test .sarif files -- say `ConverterTestData\AndroidStudio\AndroidStudio_issueLog1_raw.xml.sarif` -- by modifying the white space. Adding a single extra space should suffice.
3. Rerun the tests. It should still pass even though the string contents of the `.xml.sarif` file and the `.xml.sarif.actual` file will be different.
4. Undo your change to the test file.

Reply:
Sure. I will follow the TDD process. I have removed the test files and manually validate the changes.